core/src/main/java/org/testcontainers/containers/TestContainer.java

```diff
@@ -0,0 +1,194 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.github.dockerjava.api.model.Info;
+import lombok.NonNull;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.traits.*;
+import org.testcontainers.containers.wait.Wait;
+import org.testcontainers.containers.wait.WaitStrategy;
+import org.testcontainers.utility.SelfReference;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.time.Duration;
+import java.util.concurrent.Future;
+import java.util.function.Consumer;
+
+public interface TestContainer<SELF extends TestContainer<SELF>> extends LinkableContainer, SelfReference<SELF>, TraitsSupport<SELF>,
+        // old commands like `withExposedPorts` are supported with Support interfaces
+        Command.Support<SELF>,
```

Comment:
eventually, I would prefer to remove them in favour of `.with(new Env("key'", "value"))` notation, but, for now, it's here to simplify migration of old code.